### PR TITLE
Adding Cucumber JSON Documentation Builder

### PR DESCRIPTION
**Pull request by dgrekov**
Will throw my hat in the ring as well. I took the two previously opened and abandoned pull requests and cleaned them up as well as brought them up to the new solution layout. 

All Files are committed, all tests pass.

### CODE REVIEW

**Comment by dirkrombauts** [comment]
Thank you for your contribution! I will add some review comments to individual lines.

File: src/Pickles/Pickles.DocumentationBuilders.Cucumber/Properties/AssemblyInfo.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
+﻿using System.Reflection;
```
Comment (dirkrombauts) [inline src/Pickles/Pickles.DocumentationBuilders.Cucumber/Properties/AssemblyInfo.cs]: Please use a minimal AssemblyInfo.cs similar to the other projects, and link the VersionInfo.cs file to the project.

File: src/Pickles/Pickles.ObjectModel/ObjectModel/Feature.cs [hunk 2]
```diff
@@ -35,15 +35,15 @@ public Feature()
 
         public string Description { get; set; }
 
-        public List<IFeatureElement> FeatureElements { get; }
+        public List<IFeatureElement> FeatureElements { get; private set; }
```
Comment (dirkrombauts) [inline src/Pickles/Pickles.ObjectModel/ObjectModel/Feature.cs]: The private setter is not needed. Please remove it again.

File: src/Pickles/Pickles.ObjectModel/ObjectModel/Feature.cs [hunk 3]
```diff
@@ -35,15 +35,15 @@ public Feature()
...
 
-        public List<Comment> Comments { get; }
+        public List<Comment> Comments { get; private set; }
```
Comment (dirkrombauts) [inline src/Pickles/Pickles.ObjectModel/ObjectModel/Feature.cs]: The private setter is not needed. Please remove it again.

File: src/Pickles/Pickles.ObjectModel/ObjectModel/Feature.cs [hunk 4]
```diff
@@ -35,15 +35,15 @@ public Feature()
...
 
         public Scenario Background { get; private set; }
 
         public TestResult Result { get; set; }
 
-        public List<string> Tags { get; }
+        public List<string> Tags { get; private set; }
```
Comment (dirkrombauts) [inline src/Pickles/Pickles.ObjectModel/ObjectModel/Feature.cs]: The private setter is not needed. Please remove it again.

File: testOutput.ps1 [hunk 5]
```diff
@@ -1,5 +1,5 @@
 $version = $args[0]
-$formats = @("Html", "Dhtml", "Word", "Excel", "JSON")
+$formats = @("Html", "Dhtml", "Word", "Excel", "JSON", "Cucumber")
```
Comment (dirkrombauts) [inline testOutput.ps1]: Please add Cucumber to testRunnerCmd.cmd and testRunnerMsBuild.cmd as well.

File: src/Pickles/Pickles.DocumentationBuilders.Cucumber/Properties/AssemblyInfo.cs (hunk 1 above)
Comment (dgrekov) [inline src/Pickles/Pickles.DocumentationBuilders.Cucumber/Properties/AssemblyInfo.cs]: Done

File: testOutput.ps1 (hunk 5 above)
Comment (dgrekov) [inline testOutput.ps1]: Adjusted `testRunnerCmd.cmd` but did not find where in `testRunnerMsBuild.cmd` to make the change.

**Comment by dgrekov** [comment]
Changes made as requested, glad to be of help. Let me know if anything else should be adjusted.

File: testOutput.ps1 (hunk 5 above)
Comment (dirkrombauts) [inline testOutput.ps1]: Ah, my mistake - the output formats are defined in `testOutput.proj`, not in `testRunnersMsBuild.cmd`. Could you make the change there please?

**Comment by dirkrombauts** [comment]
Thanks again for your contribution!

**Comment by dirkrombauts** [comment]
Would you be willing (and have time) to adapt the documentation of Pickles too? It's in the [docs repository](https://github.com/picklesdoc/docs).

**Comment by dgrekov** [comment]
Sure, let me see if I have some time tonight.

**Comment by dgrekov** [comment]
Done. Please review

**Comment by dirkrombauts** [comment]
Ah, apparently approving the reviews does not automatically merge the pull request :-) I merged it now.

**Comment by dirkrombauts** [comment]
Released in version [2.12.0](https://github.com/picklesdoc/pickles/releases/tag/v2.12.0).

**Comment by bitcoder** [comment]
Can you please clarify how would you setup SpecFlow to integrate with Pickles, so pickles would output a cucumber json file?
How should the specflow project be configured? Would we have to make specflow output in Nunit XML and then use Pickles to take that XML as input, in order to convert to Cucumber JSON format?

**Comment by dgrekov** [comment]
That's basically it, let me set up a PoC and write back into this chat.

**Comment by bitcoder** [comment]
@dgrekov great, it would be great to have if you can share a working PoC :)

**Comment by dirkrombauts** [comment]
@dgrekov I think I may have found a problem with the Cucumber JSON output: it seems that for scenario outlines, there is no information about the individual examples. Are you experiencing that as well? Am I missing something?

**Comment by bitcoder** [comment]
@dgrekov were you able to set up the PoC? The online documentation is not very clear on having a full end-to-end example. I can provide you a sample .feature file and the c# class that implements the class under test. 
Will that help?
The idea was having a project in Visual Studio or Visual Code, or in other words made with .Net, using SpecFlow and where the output is Cucumber JSON file in the end.

**Comment by dgrekov** [comment]
@bitcoder, sorry for the delay, got a little behind at work. Will try for something soon.

**Comment by dgrekov** [comment]
@dirkrombauts let me look into that.

**Comment by bitcoder** [comment]
Hi @dgrekov, were you able to make a working PoC?

**Comment by dgrekov** [comment]
Sorry guys, did not mean to leave you hanging, had a hard sprint wrapping up today. Will work to get a PoC up for you in a few hours.

**Comment by dgrekov** [comment]
Got my PoC working but all of my results are coming back as inconclusive, need to look into the test harness as to see why. This will be a little bit longer. 

@bitcoder, you don't need the DLL to make this work. The only thing that is needed is the Feature file directory and the nunit XML result. I will post an example to a repo tomorrow.

**Comment by bitcoder** [comment]
ok @dgrekov, I look forward to see the example :)

**Comment by dgrekov** [comment]
Ok, I found the issue. The standard NUnit (and xUnit) report does not produce step level data, what we need is to tweak the output to include that, I think I have an approach, testing it out and will update you soon.

**Comment by bitcoder** [comment]
Hi @dgrekov , any news about this? Were you able to achieve a working solution?
thanks

**Comment by bitcoder** [comment]
Hi, were you able to solve this or make a PoC with this?

**Comment by dgrekov** [comment]
Apologies, I have been pulled into a different effort and was not able to make much progress on this. The changes needed to take advantage of actual test results was very involved and I do not think I will have time in the near future to finish it. Will post back here if I am able to contribute more.

**Pull request merged**